Cargo.toml

```diff
@@ -26,12 +26,13 @@ path = "src/lib.rs"
 regex = { version = "1.5.5", optional = true}
 serde = { version = "1.0.133", optional = true}
 serde_derive = { version = "1.0.133", optional = true}
+rand = { version = "0.8", optional = true}
```

Comment:
Let's set this to 0.8.5 as well, see https://users.rust-lang.org/t/psa-please-specify-precise-dependency-versions-in-cargo-toml/71277

Reply:
In cargo, specifying "0.y.z" means ">=0.y.z AND <0.(y+1).0". So specifying "0.8.5" gives the same flexibility as specifying "0.8", or "0.8.0", except that versions "0.8.0", "0.8.1", ..., "0.8.4" are excluded. If no other crate specifies these with a "=" or "<" dependency, i.e. "=0.8.4", no duplicate dependencies will be introduced. Specifying "=" dependencies for a well-maintained crate would be very bad practice though, so we can ignore that case.

The benefit of specifying a higher minimum than "0.8" is that we actually get all the bugfixes and features introduced between "0.8.0" and "0.8.5".

Reply:
> Specifying "=" dependencies for a well-maintained crate would be very bad practice though, so we can ignore that case.

If we ignore that case, then specifying 0.8 and 0.8.5 is the same thing, isn't it ? It's the only case in which it matters. 

The introduced feature depends on nothing introduced after 0.8.0; there is no point in lying to the package manager saying it does. The only difference it can make is that it can introduce duplicate dependencies, and there is not a single use case in which it has any positive impact. We could even use a much broader dependency specification. We use nothing more than `rand::random`, which has been stable since the very beginning of the `rand` crate.

Reply:
If we want to support rand down to 0.8.0, we need an integration test for that. There was some cargo flag in the URLO link I posted above that forces cargo to use the minimum possible dependency versions.